src/Core/ConstraintValidator/Constraints/ValidReductionValue.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\ValidReductionValueValidator;
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Constraint for validating reduction value based on reduction type
+ */
+final class ValidReductionValue extends Constraint
+{
+    public $invalidAmountValueMessage = 'Reduction value "%value%" is invalid. Value cannot be negative';
```

Comment:
This variable are not going to change, why not using const?

Reply:
Way better if constants are used :+1: . changed the code for this one.
Thank you very much @PierreRambaud